gadget/update_test.go

```diff
@@ -1588,3 +1588,146 @@ func (u *updateTestSuite) TestUpdateApplyObserverCanceledErrs(c *C) {
 
 	c.Check(logbuf.String(), testutil.Contains, `cannot observe canceled update: canceled fail`)
 }
+
+func (u *updateTestSuite) TestKernelUpdatePolicy(c *C) {
+	for _, tc := range []struct {
+		from, to *gadget.LaidOutStructure
+		update   bool
+	}{
+		{
+			from:   &gadget.LaidOutStructure{},
```

Comment:
maybe add a case that has `KernelUpdate: true` in the `from` as well to demonstrate that it doesn't matter where we are coming from, only where we are going to?